### PR TITLE
Add pyserini

### PR DESCRIPTION
**Pull request by jan-janssen**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jan-janssen** [comment]
@conda-forge/staged-recipes, ready for review

**Comment by jan-janssen** [comment]
@conda-forge/help-java @conda-forge/help-python

File: recipes/pyserini/setup.patch [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+diff --git a/setup.py b/setup.py
+index 4fb3bba27..9a9964363 100644
+--- a/setup.py
++++ b/setup.py
+@@ -7,7 +7,6 @@
+     requirements = f.read().splitlines()
+ 
+ pyserini_packages = setuptools.find_packages()
+-pyserini_packages.remove('tests')
```
Comment (ocefpaf) [inline recipes/pyserini/setup.patch:18]: This would leave a package named tests in the namespace, no?

Comment (jan-janssen) [reply recipes/pyserini/setup.patch:18]: It raises an error:
```
Processing $SRC_DIR
  Added file://$SRC_DIR to build tracker '/tmp/pip-build-tracker-z5u14kwf'
  Running setup.py (path:$SRC_DIR/setup.py) egg_info for package from file://$SRC_DIR
  Created temporary directory: /tmp/pip-pip-egg-info-suutj552
  Preparing metadata (setup.py): started
  Running command python setup.py egg_info
  Traceback (most recent call last):
    File "<string>", line 2, in <module>
    File "<pip-setuptools-caller>", line 34, in <module>
    File "/home/conda/staged-recipes/build_artifacts/pyserini_1718704178167/work/setup.py", line 10, in <module>
      pyserini_packages.remove('tests')
  ValueError: list.remove(x): x not in list
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> See above for output
```
This is why I tested building from Github directly. But then on Github the Java `*.jar` files seem to be missing, so I ended up with fixing the pip package.

Comment (ocefpaf) [reply recipes/pyserini/setup.patch:18]: That code is probably not shipped in the sdist. It should be fine then.

**Pull request merged**